Allow for specifying the chef client version in the Tyr request

lewislabs opened the pull request:
## Core change

AWS released a new version of a Ruby gem (`aws-eventstream`) which only supports Ruby >=v2.3. Chef-client uses this gem for things when provisioning a new instance. So the gem upgrade effectively borked chef-client for newly provisioned instances (via Tyr) 

v12.14 of Chef included an upgrade of Ruby and it's only a minor bump away from the version we've been using. 

This PR makes this version configurable, but defaults to the 12.14 version. 

It's been tested so far with provisioning a MongoDb router node.

## Other improvements
- Add Changelog
- Add README with pointer to docs in Sync
- Snapshot code from Hudl's `run_instance_handler` Lambda

lewislabs committed:
Allow for specifying the chef client version in the Tyr request

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/hudl/Tyr?pullRequest=154) <br/>All committers have signed the CLA.

lewislabs committed:
default to 12.14

dnordgren reviewed (approved):
TIL Tyr is open source.

Seems legit to me!

andypryor reviewed (approved):
lgtm

dnordgren committed:
NOTICK Bump version to 2.2.0 and add CHANGELOG.

dnordgren committed:
NOTICK Add README with pointer to Sync documentation.

dnordgren committed:
NOTICK Snapshot Python code from instance handler Lambda.

It's not known where this code might be kept in Hudl's GitHub repos so
snapshotting locally for ease and safekeeping.
